### DISCUSSION TITLE
Messed up EFI during update - can no longer boot from SSD or USB

### DISCUSSION THREAD

**Discussion by scottzoloft** [osy86/HaC-Mini#618] [Q&A]
Hello all,

Got my opencore NUC8 in a jam. I've been running this setup for months just fine so I know it all works (or did at some point). Here's what happened:

- i updated Mac OS to 11.5.1 and got some graphical issues. decided to try updating my lilu and whatevergreen kexts manually. EFI on SSD stopped working.
- I created a new USB using the latest HaC mini installer and an image of big sur. I was able to boot from the USB installer's EFI and get back into my SSD to back everything up and try and salvage my OS. i did notice the graphical issues went away!
- i tried to copy the initial backup i made months ago of my EFI back to my SSD's EFI partition and that did not work. USB still worked.
- i tried to save my SMBIOS info and try a fresh install. but now my NUC won't boot off my opencore USB or SSD at all. that's where i'm stuck.

Notes and things I've tried:

- updated NUC bios. set back to default and re-did the changes from the guide
- different USBs including the stick i used originally
- wiping the drive and installing windows 10 from those USBs, that works
- show menu option in the USB installer maker options
- USB stick was formatted with GUID partition table and Mac OS Journaled
- after USB installer setup, i've verified the EFI partition on the USB stick contains the boot64.efi and opencore files in their correct locations
- i can't get into opencore at all to run reset NVRAM
- i've tried the logging and verbose boot options. no log is generated, doesn't get into opencore at all.
- i've tried different combinations catalina 10.15.7 and the latest and previous installer. I've tried making the installer with a big sur image, latest and last known working.
- pressing f10 and selecting "opencore" at boot menu causes a brief flicker and no boot. letting it boot with USB as priority 1 returns a new boot device found message or boots to windows if that's on the SSD

Any assistance and suggestion are appreciated. I like this setup a lot and don't want to go back to using windows on this thing. It's been a very quick and stable system until i messed up my EFI!

EDIT: I just noticed that the USB stick is not browseable in the BIOS' EFI app browser. I see the Opencore USB Drive as a bootable ootion, but i do not see its filesystem in the browser. I just see 3 Windows partitions on SSD. I've tried other USB ports. Maybe this is meaningful?

**Comment by cgkoswund** [comment]
Maybe you can try enabling the UEFI Shell in the bios settings, then navigate manually to the EFI file?

**Enable UEFI Shell in Bios Settings:** 
Boot tab-> Boot Configuration tab -> Boot Devices section-> Internal UEFI Shell option (tick the checkbox) 

**Open Core location:**
`fs0:\EFI\OC\OpenCore.efi`

**Boot Efi location (if you just want to boot straight to the macOS installer or your SSD mac OS):**
`fs0:\EFI\BOOT\BOOTx64.efi`

It's also possible that your SSD has the right installation already, but the EFI is not picked up by the system, so you can try this:
- Switch on the PC with the SSD (leave the installer USB unplugged)
- Navigate to SSD's boot EFI manually in the UEFI shell by typing `fs0:\EFI\BOOT\BOOTx64.efi`

cd and ls commands work in the UEFI shell too, in case you get lost.

Good Luck!

**Reply by scottzoloft** [reply]
Thanks for your response! I was able to get to the Mac OS installer on USB and get going again by running opencore.efi from USB from the EFI shell on mobo. But I don't understand why that was necessary? Now that I've run opencore.efi from the EFI shell the boot entry for OpenCore is working again from my boot device selection. Do you know why that is or what the issue was?

**Reply by scottzoloft** [reply]
i'm guessing it was the issue described in the latest installer post: https://github.com/osy/HaC-Mini/releases/tag/v3.0.1

**Reply by cgkoswund** [reply]
I'm not sure why UEFI shell was necessary either! I just know that sometimes the bootable is not detected properly 😃. I believe after the first (or second?) time you use the UEFI shell, it sort of "remembers" your boot preference the next time you boot onwards, and just selects the same thing automatically...
